shared/processing/getProcessedData.ts

```diff
@@ -0,0 +1,34 @@
+import { ContentItem } from 'kentico-cloud-delivery';
+import { Operation } from '../external/models';
+import { ZapiSpecification } from '../models/zapi_specification';
+import {
+    IPreprocessedData,
+    IZapiSpecification,
+} from '../types/dataModels';
+import { processItemsInto } from './common';
+import { getApiSpecificationData } from './dataGetters';
+
+export class DataProcessor {
+    private allPreprocessedData = [];
+
+    public getProcessedData = (
+        specifications: ZapiSpecification[],
+        linkedItems: ContentItem[],
+        operation: Operation,
+    ): IPreprocessedData[] => {
+        specifications.map((item: ZapiSpecification) => {
+            this.allPreprocessedData.push(this.getDataObject(item, operation));
```

Comment:
What would happen if I called this method twice with same arguments? 😛 
I mean, what is the purpose of the private field if you never accumulate anything into it? And if you accumulated data into such class, this method should not return a value - this would be a `setter` and a new `getter` would be needed. In even other words: Why isn't this just a single "`utils`" function, why class?